public/components/trace_analytics/components/common/shared_components/component_helper_functions.tsx

```diff
@@ -51,18 +52,16 @@ export const useInjectElementsIntoGrid = (
       }
 
       const pagination = document.querySelector<HTMLElement>('.euiDataGrid__pagination');
+      const existingLoadMoreButton = pagination?.querySelector('.trace-table-load-more');
 
-      if (tracesTableMode !== 'traces') {
-        if (pagination) {
-          const existingLoadMoreButton = pagination.querySelector('.trace-table-load-more');
-          if (existingLoadMoreButton) {
-            existingLoadMoreButton.remove();
-          }
+      if (tracesTableMode !== 'traces' || !pagination || !loadMoreHandler || !isLastPage) {
```

Comment:
Is this change in the logic intentional? if so, you could remove the `pagination`, `loadMoreHandler`, and `isLastPage` from line 64, since not having one of those expressions as true would have an early return. 
Whereas I believe they were there initially because this statement here wouldn't necessarily early return for not having all those expressions/flags.

Reply:
Yes this change is intentional, I have removed the redundant check on line 64 for 
```
if (pagination && loadMoreHandler && isLastPage)
```